### ISSUE TITLE
 zigbee2mqtt  addon version 1.14.4 update breaks my system

### ISSUE THREAD

**Issue by id4vip** [danielwelch/hassio-zigbee2mqtt#378]
i am running zigbee2mqtt on raspberrypi on homeassistantOS, i have updated the zigbee2mqtt addon as well as homeassistantOS yesterday to latest version. now zigbee2mqtt is not working in addon log i am see this type of error, please guide me to resolve this error, thanks
igbee2MQTT:info  2020-09-03 12:34:08: Logging to console and directory: '/share/zigbee2mqtt/log/2020-09-03.12-34-08' filename: log.txt
Zigbee2MQTT:info  2020-09-03 12:34:11: Starting Zigbee2MQTT version 1.14.4 (commit #unknown)
Zigbee2MQTT:info  2020-09-03 12:34:11: Starting zigbee-herdsman...
Zigbee2MQTT:error 2020-09-03 12:34:14: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2020-09-03 12:34:14: Failed to start zigbee
Zigbee2MQTT:error 2020-09-03 12:34:14: Exiting...
Zigbee2MQTT:error 2020-09-03 12:34:14: Error: SREQ '--> SYS - osalNvWrite - {"id":1,"offset":0,"osal":true,"product":-1,"value":[56,234,165,28,0,75,18,0],"len":8}' failed with status '(0x02: INVALID_PARAM)' (expected '(0x00: SUCCESS)')
    at Znp.<anonymous> (/zigbee2mqtt-1.14.4/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:291:27)
    at Generator.next (<anonymous>)
    at fulfilled (/zigbee2mqtt-1.14.4/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:24:58)
npm
 ERR! code ELIFECYCLE
npm ERR! errno 1
npm
 ERR!
 zigbee2mqtt@1.14.4 start: node index.js
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.14.4 start script.
npm 
ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-09-03T18_34_14_170Z-debug.log
2020-09-03T12:34:15: PM2 log: App [npm:0] exited with code [1] via signal [SIGINT]
2020-09-03T12:34:15: PM2 log: App [npm:0] starting in -fork mode-
2020-09-03T12:34:15: PM2 log: App [npm:0] online

    zigbee2mqtt@1.14.4 start /zigbee2mqtt-1.14.4
    node index.js

**Comment by cocomaddox** [comment]
Sorry I can’t help as I have the exact same issue after upgrading. 39 devices I can no longer control as the system has broken. My error and system is the same as yours. Hope someone can help as this is the third time it has broken following an upgrade.

**Comment by id4vip** [comment]
thanks @cocomaddox, you gave me little relief as i was thinking i was only the person in problem with this system & thinking to reflash my taxes zigbee card, with your same problem i think something wrong in O/S or in Z2M addon, so i hold on to flash taxes zigbee card. also i do not want to touch that expansive card because each time with firmware flash there are chances of bricking the card. i am requesting the administrator & expert people help us to solve our problem. also i get lesson from this that there is no need to update or upgrade the system till it is not necessary as there are more chances of failure than successes. i also try to reconfigure from old known good snap shot but system not work. one thing strange I have one spare raspberry with additional working sdcard with working homeassistantOS(same system who gave me mother snapshot in past). after breaking my junior raspberry system after the update. i remove that junior raspberry system & replace with parent raspberry system(with no update). but it also having a same type of trouble. so something wrong in intersection of z2m addon & card. this is the reason  i thought may be my texas zigbee card fail. one question to z2m administrator is this addon capable to change on texas zigbee card if yes then that thing makes problem. but i prefer that card firmware be untouchable from any O/S or addon it must be standalone.   thanks

**Comment by cocomaddox** [comment]
It has happened to me three times over the past 5 weeks. Every time, the CC2531 coordinator crashes o have not been able to fix the problem. My only solution was to buy a new USB sniffer each time. I am sick and tired of doing this so I have just ordered a CC debugger app I can reflash the coordinator myself. I have searched forums everywhere and reflashing the coordinator seems to be the only valid solution. 

I am getting a bit fed up with home assistant to be honest. Every update seems to break things which take hours of troubleshooting to fix. 

Until two days ago, my system as rock solid and I thought the troubles were behind me. But no, another update breaks it again.

**Comment by id4vip** [comment]
i flashed taxes zigbee card one time last year, is it need to reflash? is it safe and chances of bricking it less. if this problem arise due to homeassistantOS, i will change my path by using normal linux & use homeassistant core. i prefer my taxes zigbee card will untouchable by any software or operating system.

**Comment by cocomaddox** [comment]
From what I understand, reflashing is the only solution. Somehow the firmware gets corrupt. I have just tried all three of my TI USBs again and they simply don’t work. Only see a red light when two days ago it was working perfectly. I am so sick of the reliability of the CC2531 that I just ordered a LAUNCHXL-CC26X2R1. Apparently they are more reliable and can handle larger networks. Fingers crossed as I doubt anyone here will help

**Comment by id4vip** [comment]
now i am feeling i am in serious trouble,  please help me to reset TI CC1352P-2 zigbee2mqtt card. this time only green led steady on, how to reset the card.
Even I thought I will reflash this card with guidelines ( https://www.zigbee2mqtt.io/information/flashing_via_uniflash.html ). Even card is not respond to uniflash software, as per guide line this card be auto detected for further process. When I put this card in my window pc with uniflash, system not able to find any device connected to it. So this card not showup in auto detect process. Please help me. so now it is little bit clear that the z2m addon update may be initiate some reset requirement for card & card is not workable till we will reset it. the software inside may be intact only need of reset this card then it will start functioning as intended. so now one million dollar question is how to reset TI CC1352P-2  card with zigbee2mqtt firmware already installed inside?

**Comment by Mattat01** [comment]
Hi,

I have the same error (copied below). I believe the problem started with the upgrade to 1.14.4 as I first noticed my zigbee devices not responding a few days ago. Restarting Hassio does not fix it, but a few days ago I pulled the power to my Pi and it came up working, only to fail again about a day later. The /share/zigbee2mqtt/log files seem to be written ever time this error comes up (every 10 secods) and only 10 log files seem to be stored, so I can't see what may have been logged when the system died. I just pulled the power to the zigbee card but it didn't reset. I power cycled the Pi again and see if it helped, but maybe it was just a coincidence last time as it is still down...

Any help in resolving or rolling back to 1.14.3 would be appreciated. 

```
Zigbee2MQTT:info  2020-09-08 09:14:52: Logging to console and directory: '/share/zigbee2mqtt/log/2020-09-08.09-14-52' filename: log.txt
Zigbee2MQTT:info  2020-09-08 09:14:55: Starting Zigbee2MQTT version 1.14.4 (commit #unknown)
Zigbee2MQTT:info  2020-09-08 09:14:55: Starting zigbee-herdsman...
Zigbee2MQTT:error 2020-09-08 09:14:56: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2020-09-08 09:14:56: Failed to start zigbee
Zigbee2MQTT:error 2020-09-08 09:14:56: Exiting...
Zigbee2MQTT:error 2020-09-08 09:14:56: Error: Error while opening serialport 'Error: Error: Resource busy, cannot open /dev/ttyACM0'
    at Znp.<anonymous> (/zigbee2mqtt-1.14.4/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:136:32)
    at Generator.next (<anonymous>)
    at /zigbee2mqtt-1.14.4/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:27:71
    at new Promise (<anonymous>)
    at __awaiter (/zigbee2mqtt-1.14.4/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:23:12)
    at SerialPort.<anonymous> (/zigbee2mqtt-1.14.4/node_modules/zigbee-herdsman/dist/adapter/z-stack/znp/znp.js:134:49)
    at SerialPort._error (/zigbee2mqtt-1.14.4/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:198:14)
    at /zigbee2mqtt-1.14.4/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:242:12
npm
 ERR! code ELIFECYCLE
npm ERR! errno 1
npm
 ERR! zigbee2mqtt@1.14.4 start: `node index.js`
npm 
ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.14.4 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR!
 A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2020-09-07T23_44_56_452Z-debug.log
2020-09-08T09:14:59: PM2 log: App [npm:0] exited with code [1] via signal [SIGINT]
2020-09-08T09:14:59: PM2 log: App [npm:0] starting in -fork mode-
2020-09-08T09:14:59: PM2 log: App [npm:0] online
```

**Comment by id4vip** [comment]
in last two days i try homeassistantOS on (raspberrypi-2 & raspberry pi-3) also rasbian lite  on raspberrypi-3 with "TI CC1352P-2 zigbee2mqtt" card. still zigbee2mqtt addon fail to work also on rasbian lite after the command "npm start" according to guidelines system again fail  ( https://www.zigbee2mqtt.io/getting_started/running_zigbee2mqtt.html ) so still my system is down  after upgrade with version-1.14.4

**Comment by id4vip** [comment]
i just reflash my  "TI CC1352P-2" & after that insert that card into raspberry pi & all my sensor again online. so the reflashing of card is the only solution. thanks

**Comment by cocomaddox** [comment]
It does appear to be the only solution. Odd that the coordinators become. On responsive following a software update. Maybe I just won’t update my software anymore or will look for a more stable solution. I can’t reflash my devices until my debugger arrives.

**Comment by Mattat01** [comment]
I'll try reflashing later, but I'd be surprised if we all had our hardware fail at the same time as an update. Maybe the update relies on newer firmware on the coordinator...?

**Comment by cocomaddox** [comment]
My hardware has failed three times over the past few months. Strange coincidence that it failed following every software update. I had my updates set to occur automatically. I am going to revert to manual updates now so I can stop the Zigbee2mqtt add on before I run the update. Don’t know if it the cause but I really don’t know what else it can be. Too much of a coincidence for me.

**Comment by Mattat01** [comment]
I'm wierded out by this. Reflashing the coordinator and routers did actually fix it. I had/have to reconnect every device but the config was still there when I re-paired them, so that's good. I still don't get how a software upgrade kills the coordinator/router so maybe there is some other problem that the reflashing just fixes, like a break in the link between the coordinator and Home Assistant, so the fresh flash causes a re-pair. Regardless, it worked and now I believe you all :)
Thanks.

**Comment by cocomaddox** [comment]
I also reflashed my coordinator last night. Has to repair 39 devices but like you, the config And customisation was still there. I have Ben through this many times and have done a lot of research on it via forums. It seems the only solution, every time, is to reflash the firmware. 

I have no idea that causes the firmware to become corrupt but I can tell you that I will not be updating Add on software again as I am sick of my system breaking as a result.

**Comment by dseifert** [comment]
Same issue here, after updating to 1.14.4 I had to re-flash the firmware (CC2531ZNP-Prod.hex). I wonder whether it's related to https://github.com/Koenkk/zigbee2mqtt/issues/3890

**Comment by Mattat01** [comment]
@dseifert it could be related. Whatever caused it damaged the stick because my configuration still didn't work after rolling back to my back-up. I can't believe that it was just a stability issue thoug, as I had to reflash my coordinator and then reflash both routers to get them all back on line. Three wouldn't all fail at once (but to be fair I didn't try too hard to reconnect them before flashing).

Anyway, it remains a mystery to me...

**Comment by cocomaddox** [comment]
I received my new LAUNCHXL-CC26X2R1 a few days ago. Flashed the device and it has been running for 5 days without a single error in my logs. I have been reluctant to try to update Home Assistant or the zigbee2mqtt add-on yet but will try this soon. Compared to the CC2531, this device is a lot more powerful and stable. The logs, when I was running the CC2531, were full of connection errors. I would never recommend using the CC2531 as it seems to be a low powered, unreliable device that crashes every time the add on was updated (at least it did for me). I had to replace my CC2531 three times over the past two months (as I am still waiting for the debugger to arrive from China).

Its a mystery to me too, but for anyone new to zigbee2mqtt, buy the LAUNCHXL-CC26X2R1 and forget about using low powered devices.
